testfixtures/comparison.py

```diff
@@ -685,3 +685,27 @@ def diff(x, y, x_label='', y_label=''):
             y_label or 'second',
             lineterm='')
     )
+
+
+class RangeComparison:
+    """
+    An object that can be used in comparisons of numerics to validate
+    actual falls within a certain range
+    """
+    def __init__(self, lower_bound, upper_bound):
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+
+    def __eq__(self, other):
+        try:
+            0 + other
```

Comment:
If you want to be more strict and make sure comparison against a non-numeric type raises a `TypeError` you could check explicitly using `isinstance(other, numbers.Number)` however this is arguably un-Pythonic.

The consequence of not doing this strict check is not significant: The range comparison will quietly return False, which would then raise a suitable error message about the failed comparison, which should make it obvious that a non-numeric type was passed in for comparison. That seems adequate...